libs/tendermint/crypto/encoding/codec.go

```diff
@@ -22,27 +23,42 @@ func PubKeyToProto(k crypto.PubKey) (pc.PublicKey, error) {
 				Ed25519: k[:],
 			},
 		}
+	case secp256k1.PubKeySecp256k1:
+		kp = pc.PublicKey{
+			Sum: &pc.PublicKey_Secp256K1{
+				Secp256K1: k[:],
+			},
+		}
 	default:
 		return kp, fmt.Errorf("toproto: key type %v is not supported", k)
 	}
 	return kp, nil
 }
 
 // PubKeyFromProto takes a protobuf Pubkey and transforms it to a crypto.Pubkey
-func PubKeyFromProto(k *pc.PublicKey) (crypto.PubKey, error) {
+// Return one more parameter to prevent of slowing down the whole procedure
+func PubKeyFromProto(k *pc.PublicKey) (crypto.PubKey, bool, error) {
```

Comment:
bool can defined to (isSpec bool) , then set and return isSpec true or false,  for clearer.

Reply:
or use enum